utils/das_data_sorter.py

```diff
@@ -1,6 +1,79 @@
 import pandas as pd 
 import argparse 
 from math import ceil
+from numpy import median, mean
+
+# averages data within the seconds or minutes, according to new_time array
+def avg_Data(array):
+    avg_array = []
+
+    for millisecond_indices in new_time:
```

Comment:
Here we're accessing the variable `new_time`, which isn't defined until much later in the file. Python allows this, but in most cases I'd try to avoid accessing variables outside of the scope of the function.

The reason for this is mainly because it makes code easier to read, and easier to debug. When I first read this code, I had no clue what this variable was, but I would have a better idea if it was instead a parameter of the function. As for debugging, functions are much easier to debug when they are deterministic - meaning that for the same inputs, they will always return the same outputs. This is not the case here, as the function's output also depends on the value of `new_time` which is completely independent.

`new_time` appears in a few other functions around this file, they could be made parameters too!